Apps/W1/HybridGP/app/src/Migration/GPTables/GPPopulateCombinedTables.Codeunit.al

```diff
@@ -763,38 +763,42 @@ codeunit 40125 "GP Populate Combined Tables"
     internal procedure PopulateGPItemTransactions()
     var
         GPItemTransactions: Record "GP Item Transactions";
+        GPIV00101: Record "GP IV00101";
+        Item: Record Item;
         GPPopulateItemTransactions: Query "GP Populate Item Transactions";
     begin
         GPPopulateItemTransactions.SetRange(RCPTSOLD, false);
         GPPopulateItemTransactions.SetRange(QTYTYPE, 1);
         GPPopulateItemTransactions.Open();
-        while GPPopulateItemTransactions.Read() do begin
-            Clear(GPItemTransactions);
-            GPItemTransactions.No := CopyStr(GPPopulateItemTransactions.ITEMNMBR.Trim(), 1, MaxStrLen(GPItemTransactions.No));
-            GPItemTransactions.Location := CopyStr(GPPopulateItemTransactions.TRXLOCTN.Trim(), 1, MaxStrLen(GPItemTransactions.Location));
-            GPItemTransactions.DateReceived := GPPopulateItemTransactions.DATERECD;
-            GPItemTransactions.UnitCost := GPPopulateItemTransactions.UNITCOST;
-            GPItemTransactions.ReceiptSEQNumber := GPPopulateItemTransactions.RCTSEQNM;
-
-            case GPPopulateItemTransactions.ITMTRKOP of
-                2:
-                    GPItemTransactions.Quantity := 1;
-                3:
-                    GPItemTransactions.Quantity := GPPopulateItemTransactions.QTYRECVDGPIV00300 - GPPopulateItemTransactions.QTYSOLDGPIV00300;
-                else
-                    GPItemTransactions.Quantity := GPPopulateItemTransactions.QTYRECVD - GPPopulateItemTransactions.QTYSOLD;
-            end;
+        while GPPopulateItemTransactions.Read() do
+            if GPIV00101.Get(GPPopulateItemTransactions.ITEMNMBR) then
+                if ShouldAddItemToStagingTable(GPIV00101) then begin
+                    Clear(GPItemTransactions);
+                    GPItemTransactions.No := CopyStr(GPPopulateItemTransactions.ITEMNMBR.Trim(), 1, MaxStrLen(Item."No."));
+                    GPItemTransactions.Location := CopyStr(GPPopulateItemTransactions.TRXLOCTN.Trim(), 1, MaxStrLen(GPItemTransactions.Location));
+                    GPItemTransactions.DateReceived := GPPopulateItemTransactions.DATERECD;
+                    GPItemTransactions.UnitCost := GPPopulateItemTransactions.UNITCOST;
+                    GPItemTransactions.ReceiptSEQNumber := GPPopulateItemTransactions.RCTSEQNM;
+
+                    case GPPopulateItemTransactions.ITMTRKOP of
+                        2:
```

Comment:
Minor - it would be good to have comments about what 2 and 3 mean

Reply:
we could optionally introduce an enum

Reply:
I added comments, thanks.